Source/CarthageKit/Git.swift

```diff
@@ -217,14 +227,24 @@ public func cloneSubmoduleInWorkingDirectory(_ submodule: Submodule, _ workingDi
 				}
 		}
 
-	return SignalProducer<(), CarthageError> { () -> Result<(), CarthageError> in
-		repositoryCheck("remove submodule checkout") {
-			try FileManager.default.removeItem(at: submoduleDirectoryURL)
+	let dependencyForSubmodule = Dependency.git(submodule.url)
+
+	return cloneOrFetchOnComplete(dependency: dependencyForSubmodule, preferHTTPS: preferHTTPS, commitish: submodule.sha)
```

Comment:
`Git.swift` is meant to be independent of the Carthage-specifics, so I don't think we should call `cloneOrFetch` here. If we need to do something like this here, then we should add a `referenceRepository: URL?` parameter to this function to keep it generic.

Reply:
How can it be possible when this `cloneSubmoduleInWorkingDirectory` function deals with all submodules recursively? Pass a map that maps any submodule to its cache path?